axum-jwks/src/key_store.rs

```diff
@@ -0,0 +1,155 @@
+use std::{collections::HashMap, str::FromStr, time::Duration};
+
+use jsonwebtoken::{
+    jwk::{self, AlgorithmParameters},
+    DecodingKey, Validation,
+};
+use serde::Deserialize;
+use thiserror::Error;
+
+use tokio::time::Instant;
+
+const DEFAULT_ALG: jsonwebtoken::Algorithm = jsonwebtoken::Algorithm::RS256;
+
+type Keys = HashMap<String, Jwk>;
+
+#[derive(Clone)]
+pub(crate) struct KeyStore {
+    pub last_updated: Instant,
+    pub keys: Keys,
+}
+
+impl Default for KeyStore {
+    fn default() -> Self {
+        Self {
+            last_updated: Instant::now() - Duration::from_secs(3600 * 24 * 365),
```

Comment:
I think making this an `Option` instead of faking times in the past makes more sense.

Reply:
Agreed, after a little tweaking I got the usage to be ok after all.